earth_enterprise/rpms/pre-install_fusion.sh

```diff
@@ -0,0 +1,167 @@
+#!/bin/bash
+#
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# NOTE: requires xmllint from libxml2-utils
+
+set +x
+
+#-----------------------------------------------------------------
+BADHOSTNAMELIST=(empty linux localhost dhcp bootp)
+
+# versions and user names
+GEE="Google Earth Enterprise"
+GEES="$GEE Server"
+
+# directory locations
+BININSTALLROOTDIR="/etc/init.d"
+BASEINSTALLDIR_OPT="/opt/google"
+BASEINSTALLDIR_ETC="/etc/opt/google"
+BASEINSTALLDIR_VAR="/var/opt/google"
+
+# derived directories
+SYSTEMRC="$BASEINSTALLDIR_ETC/systemrc"
+
+# Get system info values
+NEWLINECLEANER="sed -e s:\\n::g"
+HOSTNAME="$(hostname -f | tr [A-Z] [a-z] | $NEWLINECLEANER)"
+HOSTNAME_F="$(hostname -f | $NEWLINECLEANER)"
+
+check_bad_hostname() {
+    if [ -z "$HOSTNAME" ] || [[ " ${BADHOSTNAMELIST[*]} " == *"${HOSTNAME,,} "* ]]; then
+        show_badhostname
+
+        if [ $BADHOSTNAMEOVERRIDE == true ]; then
+            echo -e "Continuing the installation process...\n"
+            # 0 = true (in bash)
+            return 0
+        else
+            echo -e "Exiting the installer.  If you wish to continue, re-run this command with the -hnf 'Hostname Override' flag.\n"
+            # 1 = false
+            return 1
+        fi
+    fi
+}
+
+show_badhostname()
+{
+    echo -e "\nYour server [$HOSTNAME] contains an invalid hostname value which typically"
+    echo -e "indicates an automatically generated hostname that might change over time."
+    echo -e "A subsequent hostname change would cause configuration issues for the "
+    echo -e "$SOFTWARE_NAME software.  Invalid values: ${BADHOSTNAMELIST[*]}."
+}
+
+check_mismatched_hostname() {
+    if [ $HOSTNAME != $HOSTNAME_F ]; then
+        show_mismatchedhostname
+
+        if [ $MISMATCHHOSTNAMEOVERRIDE == true ]; then
```

Comment:
Same comment. We should assume `MISMATCHEDHOSTNAMEOVERRIDE` is always `true` and remove this section.